components/places/android/src/main/java/org/mozilla/places/PlacesConnection.kt

```diff
@@ -286,6 +355,33 @@ interface PlacesAPI {
      */
     fun getVisited(urls: List<String>): List<Boolean>
 
+    /**
+     * Returns a list of visited URLs for a given time range.
+     *
+     * @param start beginning of the range, unix timestamp in milliseconds.
+     * @param end end of the range, unix timestamp in milliseconds.
+     * @param includeRemote boolean flag indicating whether or not to include remote visits. A visit
+     *  is (roughly) considered remote if it didn't originate on the current device.
+     */
+    fun getVisitedUrlsInRange(start: Long, end: Long = Long.MAX_VALUE, includeRemote: Boolean = true): List<String>
+
+    /**
+     * Get detailed information about all visits that occurred in the
+     * given time range.
+     *
+     * @param start The (inclusive) start time to bound the query.
+     * @param end The (inclusive) end time to bound the query.
+     */
+    fun getVisitInfos(start: Long, end: Long = Long.MAX_VALUE): List<VisitInfo>
+}
+
+interface WritablePlacesConnectionInterface {
```

Comment:
I feel like this should extend ReadablePlacesConnectionInterface.

E.g. `PlacesConnectionInterface` > `ReadablePlacesConnectionInterface` > `WritablePlacesConnectionInterface` seems reasonable to me, and is what we have in practice.